src/main/java/seedu/address/model/task/Task.java

```diff
@@ -1,29 +1,71 @@
 package seedu.address.model.task;
 
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+import seedu.address.model.tag.Tag;
+
+
 /**
  * Represents a Task in the TaskList.
+ * Guarantees: description is present and not null, field values are validated, immutable.
  */
 public class Task {
-    private Description description;
-    private boolean isDone;
+    private final Description description;
+    private final Deadline deadline;
+    private Boolean isDone;
+    private final Set<Tag> tags = new HashSet<>();
+
 
     /**
      * A constructor that creates an instance of Task.
      * @param description The description of the task.
      */
-    public Task(Description description) {
-        this.isDone = false;
+    public Task(Description description, Deadline deadline, Boolean isDone, Set<Tag> tags) {
+        requireAllNonNull(description, deadline, isDone, tags);
```

Comment:
Maybe we can allow for null values for deadline, for tasks that don't have a deadline? But we can do this in a later iteration

Reply:
yes, currently it accepts empty deadlines (deadline is empty string). The non-null is so that the deadline object in task is not equals to null